runtime/v2/shim/util.go

```diff
@@ -64,21 +64,28 @@ func Command(ctx context.Context, runtime, containerdAddress, containerdTTRPCAdd
 		cmdPath = cmdPathI.(string)
 	} else {
 		var lerr error
-		if cmdPath, lerr = exec.LookPath(name); lerr != nil {
-			if eerr, ok := lerr.(*exec.Error); ok {
-				if eerr.Err == exec.ErrNotFound {
-					// LookPath only finds current directory matches based on
-					// the callers current directory but the caller is not
-					// likely in the same directory as the containerd
-					// executables. Instead match the calling binaries path
-					// (containerd) and see if they are side by side. If so
-					// execute the shim found there.
-					testPath := filepath.Join(filepath.Dir(self), name)
-					if _, serr := os.Stat(testPath); serr == nil {
-						cmdPath = testPath
-					}
-					if cmdPath == "" {
-						return nil, errors.Wrapf(os.ErrNotExist, "runtime %q binary not installed %q", runtime, name)
+		binaryPath := BinaryPath(runtime)
+		if _, serr := os.Stat(binaryPath); serr == nil {
```

Comment:
os.Stat works for relative paths as well no? how do we test `binaryPath` is indeed a "full binary path"

Reply:
@dims, first of all, thanks for the very quick review.

Please, take a look at the newly introduced [`BinaryPath()`](https://github.com/fidencio/containerd/blob/d80dbdae6823804cae096dec41542ec68145e0cd/runtime/v2/shim/util.go#L133-L145).

There I'm getting the joining the $prefix and the binary name, and returning its absolute path.  This is enough to take the full path into consideration, right?

Or your consideration is that `filepath.Abs()` returns also a relative path and using the relative path could be problematic?

Reply:
Ah i missed that. thanks! (the logic in the BinaryPath)